### ISSUE TITLE
[alert] Alerts inside a modal can queue behind overlay

### ISSUE THREAD

**Issue by nwhittaker** [Esri/calcite-design-system#6818]
### Actual Behavior

If an open alert exists before opening a modal, any alerts opened within the modal are queued within the original alert. However the original alert is inaccessible behind the modal overlay.

<img width="839" alt="Screenshot 2023-04-20 at 3 30 27 PM" src="https://user-images.githubusercontent.com/421496/233470356-ad37762a-ddb9-4123-ae00-7ee34e441dfe.png">

### Expected Behavior

Any alerts opened within the modal should probably start their own alert queue above the modal overlay. Closing the modal should restore/reveal the originally opened alert.

<img width="839" alt="Screenshot 2023-04-20 at 3 30 44 PM" src="https://user-images.githubusercontent.com/421496/233470416-0e7c4e2a-3350-4693-ba80-00983a9f6fde.png">

### Reproduction Sample

https://codepen.io/nwhittaker-esri/pen/LYgRWZR

### Reproduction Steps

1. View code sample and see alert queue with both alerts is obscured in background.

### Reproduction Version

1.2.0

### Priority impact

p3 - want for upcoming milestone

### Impact

Our modals can have complex UIs where it's useful to use alerts to notify the user of the results of various operations within the modal.

A workaround is to programmatically close all alerts before opening a modal. It's a bit destructive, though, and not something the user might expect to happen.

### Esri team

ArcGIS Field Apps

**monday.com sync:** #9357359820

**Comment by macandcheese** [comment]
Currently Alerts are positioned fixed to the page, so the Modal / Scrim will always be blocking. We could add an `alerts` slot to Modal, and position the Alerts relative to Modal, although they'd likely make more sense from a layering perspective to be positioned with the Modal than "next to it" above the Scrim.

Currently it's not possible to have grouped queues, so any Alert on the page are in the same queue. With #2835, this might be a good enhancement at the same time.

**Comment by nwhittaker** [comment]
>Currently Alerts are positioned fixed to the page, so the Modal / Scrim will always be blocking.

I'm not sure what you mean by blocking. Alerts inserted into a modal are positioned above the modal because they inherit the modal's stacking context: https://codepen.io/nwhittaker-esri/pen/vYVePPJ.

<img width="843" alt="Screenshot 2023-05-02 at 2 50 08 PM" src="https://user-images.githubusercontent.com/421496/235758222-90e26f6f-0c47-40f5-b765-a9ed9c2a1535.png">

**Comment by macandcheese** [comment]
Oh, I missed the `slot="content"` on the Alert in Codepen, apologies.

**Comment by geospatialem** [comment]
Updated to an enhancement request to consider the use case. In the meantime, a `z-index` can be added to accommodate the use case. For instance: https://codepen.io/mac_and_cheese/pen/ZEmRGQO

**Comment by jcfranco** [comment]
I [updated the above codepen](https://codepen.io/jcfranco/pen/YzRvXJB) to use our token CSS vars, which are available since 1.4.0.

**Comment by mpayson** [comment]
+ 1. It seems like this is becoming a common pattern but it is hard to implement and position the alert correctly. Currently there are custom implementations in Map Viewer, Home app, Data Pipelines... so for us (Data Pipelines) it'd be a great enhancement

Would it be possible to implement in two phases, one to add the slot with a note about queuing, one to manage multiple queues?

**Comment by macandcheese** [comment]
@nwhittaker better late than never... could you confirm if the (not-so-recently-anymore) added [Alerts slot](https://developers.arcgis.com/calcite-design-system/components/dialog/#api-reference-slots-alerts) in Dialog (and [in Panel](https://developers.arcgis.com/calcite-design-system/components/panel/#api-reference-slots-alerts)) is sufficient for this use case: https://codepen.io/mac_and_cheese/pen/wBzpmJj?editors=1000

**Comment by nwhittaker** [comment]
>@nwhittaker better late than never... could you confirm if the (not-so-recently-anymore) added [Alerts slot](https://developers.arcgis.com/calcite-design-system/components/dialog/#api-reference-slots-alerts) in Dialog (and [in Panel](https://developers.arcgis.com/calcite-design-system/components/panel/#api-reference-slots-alerts)) is sufficient for this use case: https://codepen.io/mac_and_cheese/pen/wBzpmJj?editors=1000

@macandcheese yes, I think the Popover API conversion work and `alerts` slot resolve my original layering issue -- thanks!

**Comment by macandcheese** [comment]
Closing based on above comment cc @Esri/calcite-pes
